src/freenet/client/async/SplitFileFetcher.java

```diff
@@ -297,7 +298,10 @@ public void onSplitfileCompatibilityMode(CompatibilityMode min, CompatibilityMod
     @Override
     public void queueHeal(byte[] data, byte[] cryptoKey, byte cryptoAlgorithm) {
         try {
-            context.healingQueue.queue(BucketTools.makeImmutableBucket(context.tempBucketFactory, data), cryptoKey, cryptoAlgorithm, context);
+            RandomAccessBucket dataBucket = BucketTools.makeImmutableBucket(
```

Comment:
Is `RandomAccessBucket` necessary here? `SimpleHealingQueue.queue` seems to accept a `Bucket`, so we shouldn’t be handling more than `Bucket`s.